src/EssentialSigner/index.ts

```diff
@@ -1,30 +1,45 @@
 import {
-  BlockTag,
   Provider,
   TransactionRequest,
   TransactionResponse,
 } from '@ethersproject/abstract-provider';
+import { ExternallyOwnedAccount } from '@ethersproject/abstract-signer';
 import { Deferrable, defineReadOnly } from '@ethersproject/properties';
-import { logger, Signer } from 'ethers';
-import { Bytes, Logger, resolveProperties } from 'ethers/lib/utils';
+import { BigNumberish, logger, Signer, Wallet } from 'ethers';
+import { Bytes, Logger } from 'ethers/lib/utils';
 
-import { EIP712Domain, EIP712StructField } from './messageSigner';
+import EssentialForwarder from '../abi/EssentialForwarder.json';
+import {
+  EIP712Domain,
+  EIP712StructField,
+  signMetaTxRequest,
+} from './messageSigner';
+
+export interface EssentialOverrides {
+  authorizer: string;
+  nftContract: string;
+  nftChainId: BigNumberish;
+  nftTokenId: BigNumberish;
+}
 
-export class EssentialSigner extends Signer {
+export class EssentialSigner extends Signer implements ExternallyOwnedAccount {
   readonly address: string;
+  readonly privateKey: string;
   readonly provider: Provider;
   readonly relayerUri: string;
 
   constructor(
     address: string,
-    provider: Provider,
+    provider?: Provider,
+    wallet?: Wallet,
     // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
     relayerUri: string = process.env.RELAYER_URI!,
```

Comment:
im never really quite sure how to handle things like this in a JS lib? its like a global configuration - would prefer not to pass it in to every constructor call?